htdocs/core/lib/pdf.lib.php

```diff
@@ -1541,6 +1542,10 @@ function pdf_getlinedesc($object, $i, $outputlangs, $hideref = 0, $hidedesc = 0,
 		}
 	}
 
+	if (isModEnabled('barcode') && getDolGlobalString('MAIN_GENERATE_DOCUMENTS_SHOW_PRODUCT_BARCODE') && !empty($product_barcode)) {
```

Comment:
May be you can do 
$libelleproduitservice = dol_concatdesc($libelleproduitservice, $langs->trans("BarCode")." ".$product_barcode);
So it is clear for PDF reader that this number alone on the line is a barcode.

Reply:
@eldy You're right! that will be more clear, I added the barcode label